.env-example

```diff
@@ -0,0 +1,23 @@
+# Required redis host for backing the queues for Machinery
+MACHINERY_REDIS_HOST=redis://localhost:6379
+
+# Optional MongoDB Backing for task state
+MACHINERY_MONGODB_HOST=mongodb://root:abcd1234@localhost:27017
+
+# Expire time for mongodb backing for task state
+MACHINERY_MONGODB_EXPIRE_TIME_SECONDS=3600
+
+# If the service has workers, enable it here. Otherwise a service may only submit tasks
+MACHINERY_WORKERS_ENABLED=true
+
+# Alias of the worker for logs. Could be unique podname, etc.
+MACHINERY_WORKER_NAME=example-tasks
+# Max amount of tasks to process in parallel. 0 = infinite
+MACHINERY_WORKER_CONCURRENCY_LIMIT=0
+
+# Number of times to retry a task if it errors
+MACHINERY_RETRY_ATTEMPTS=100
```

Comment:
action - too much for a default value. should be something like 3 to 5?

Reply:
The events will be deleted in the queue after 3-5, so if we do not have retry, its better not to use queues because we will end up with the same problem (event fails, manual intervention)....

Queues should be used for 100% guarantee, stay in the queue indefinitely until someone consumes the event successfully.

to discuss during PR meeting.

Reply:
umm... not agree that it is for 100% guarantee. there are so many cases that the task can fail. So the task queue to me is to
1. guarantee the chained task is guaranteed all-or-nothing
  -> so if one of the tasks is failed unavoidably, we need to have either logged them with alerts or even fire another chained task for recovery
2. parallel workers consuming tons of tasks

infinite retrying will block the workers from processing other tasks...


And one point to make sure. The failed task u mean it is deleted from the queue but it should be stored somewhere else right? (in Mongo I guess?)

Reply:
Create Account -> Create Stripe Account (uuid) -> Create Subscription (uuid) -> Store results from step 2+3.

- CreateStripeAccount
- CreateSubscription
- ChargeSubscription
- Webhooks (Git commits)

Reply:
agree to have a super-high retry count since we are using the message queue for the mission-critical task for users. and we need to make sure the tasks are persisted to disk.